packages/x-date-pickers/src/AdapterDateFnsJalali/AdapterDateFnsJalali.ts

```diff
@@ -472,19 +472,19 @@ export class AdapterDateFnsJalali implements MuiPickersAdapter<Date, DateFnsLoca
   };
 
   public getWeekArray = (value: Date) => {
-    const start = startOfWeek(startOfMonth(value), { locale: this.locale });
-    const end = endOfWeek(endOfMonth(value), { locale: this.locale });
+    const start = this.startOfWeek(this.startOfMonth(value));
+    const end = this.endOfWeek(this.endOfMonth(value));
 
     let count = 0;
     let current = start;
     const nestedWeeks: Date[][] = [];
 
-    while (isBefore(current, end)) {
+    while (this.isBefore(current, end)) {
       const weekNumber = Math.floor(count / 7);
       nestedWeeks[weekNumber] = nestedWeeks[weekNumber] || [];
       nestedWeeks[weekNumber].push(current);
 
-      current = addDays(current, 1);
+      current = this.addDays(current, 1);
       count += 1;
     }
```

Comment:
Do you need to keep it, or overriding the `startOfWeek` is enought to keep the definition of `getWeekArray` from `AdapterDateFns`?

Reply:
Ok, better not to try that :)

Reply:
It would be cool to be able to do that, but given the low adoption of this adapter it's clearly not worth the effort :laughing: